src/jit/codegenlinear.cpp

```diff
@@ -1297,6 +1297,10 @@ void CodeGen::genConsumeRegs(GenTree* tree)
         {
             genConsumeReg(tree->gtGetOp1());
         }
+        else if (tree->OperIsHWIntrinsic())
```

Comment:
It seems like this would be a good place to assert that `HW_Flag_NoContainment` is not set, or that `IsContainableHWIntrinsicOp(tree, tree->gtGetOp1())` but perhaps that's overkill.

Reply:
I think asserting is a good idea. I'll update.

This is also pending https://github.com/dotnet/coreclr/pull/16114 and https://github.com/dotnet/coreclr/pull/16115. The former to fixup the various flags and the latter to make it easier to add some tests with this change (we currently don't have any tests that use `Load`, `LoadScalar`, or `LoadAligned` in a way that they could be contained).

Reply:
Fixed.

Reply:
Hmm, actually, I can't easily check `IsContainableHWIntrinsicOp`.

It is a member of `Lowering` and depends on `IsContainableMemoryOp`, which itself depends on `m_lsra`. 

I would need to make `m_pLowering` visible from `compiler` somewhere to use that check.

The other check, `flagsOfHWIntrinsic`, currently requires friend access to the compiler, but I don't see any reason why those can't be public.

@CarolEidt, what would you recommend here?

Reply:
I can't recall why `IsContainableMemoryOp` has its actual implementation on `LinearScan` when all the containment analysis is done in `Lowering` (even before we moved the `TreeNodeInfoInit` to LSRA). In any case, it seems like something we might want to check during code generation, so we may want to expose it in some way. That said, I don't feel that it's critical.

On the other hand, the `flagsOfHwIntrinsic` I think should definitely be public so that they are accessible from codegen.

Reply:
> I can't recall why IsContainableMemoryOp has its actual implementation on LinearScan when all the containment analysis is done in Lowering (even before we moved the TreeNodeInfoInit to LSRA).

Looks to be one use in LSRA, for `SIMDIntrinsicGetItem`: https://github.com/dotnet/coreclr/blob/master/src/jit/lsraxarch.cpp#L3026

> In any case, it seems like something we might want to check during code generation, so we may want to expose it in some way. That said, I don't feel that it's critical.

Exposing a `getLowering()` method in the compiler is easy enough, but it also requires us to `#include "lower.h"` somewhere where `codegen` can pick it up (probably just in `codegen.h`). I've commented out this particular assert with a TODO and will finish looking tomorrow if I get the chance.

> On the other hand, the flagsOfHwIntrinsic I think should definitely be public so that they are accessible from codegen.

It is already available to codgen, I just forgot to put the assert in an `#ifdef _TARGET_XARCH_` since its in `codegenlinear`. 😄

Reply:
Both asserts I tried to add here would have been invalid. `tree` is the node which was contained, we would want to be doing the assert on the node which contains `tree`.

Reply:
Moved the asserts to `hwintrinsiccodegenxarch.cpp`.